composer.json

```diff
@@ -16,16 +16,16 @@
         "issues": "https://github.com/aws/aws-sdk-php/issues"
     },
     "require": {
-        "php": ">=5.5",
-        "guzzlehttp/guzzle": "^6.5.8 || ^7.4.5",
-        "guzzlehttp/psr7": "^1.9.1 || ^2.4.5",
-        "guzzlehttp/promises": "^1.4.0",
+        "php": ">=7.2.5",
+        "guzzlehttp/guzzle": "^7.4.5",
+        "guzzlehttp/psr7": "^2.4.5",
```

Comment:
we don't need to drop 1.x here. we can have `^1.9.1 || ^2.4.5`. having extra return types in your implementations is still compatible with v1.